src/auto/gradient.rs

```diff
@@ -0,0 +1,31 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir)
+// from gir-files (https://github.com/gtk-rs/gir-files)
+// DO NOT EDIT
+
+use cairo;
+use glib::object::IsA;
+use glib::translate::*;
+use gtk_sys;
+use StyleContext;
+
+glib_wrapper! {
+    #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+    pub struct Gradient(Boxed<gtk_sys::GtkGradient>);
+
+    match fn {
+        copy => |ptr| gobject_sys::g_boxed_copy(gtk_sys::gtk_gradient_get_type(), ptr as *mut _) as *mut gtk_sys::GtkGradient,
+        free => |ptr| gobject_sys::g_boxed_free(gtk_sys::gtk_gradient_get_type(), ptr as *mut _),
+        get_type => || gtk_sys::gtk_gradient_get_type(),
+    }
+}
+
+impl Gradient {
+    pub fn resolve_for_context<P: IsA<StyleContext>>(&self, context: &P) -> Option<cairo::Pattern> {
```

Comment:
The whole type is deprecated since 3.8, just in this function it was forgotten to specify the version